src/views/components/MoreSettings.jsx

```diff
@@ -66,9 +69,15 @@ export const MoreSettings = () => {
 
   const [meetingURL, setMeetingURL] = useState(defaultMeetingUrl);
   const [rtmpURL, setRtmpURL] = useState("");
+  const [isHlsOn, setIsHlsOn] = useState(false);
   const recording = useHMSStore(selectRecordingState);
   const rtmp = useHMSStore(selectRTMPState);
+  const hls = useHMSStore(selectHLSState);
+  const [prevRole, setPrevRole] = useState(localPeer.roleName);
   const [isRecordingOn, setIsRecordingOn] = useState(false);
+  const HLSRole =
+    roomMetadata[DEFAULT_HLS_ROLE_KEY] ||
+    process.env.REACT_APP_DEFAULT_HLS_ROLE;
```

Comment:
The redundant env variable can be removed. Room metadata will already have the role